llvm/include/llvm/TableGen/StringToOffsetTable.h

```diff
@@ -52,6 +53,35 @@ class StringToOffsetTable {
     return II->second;
   }
 
+  // Emit the string using string literal concatenation, for better readability
+  // and searchability.
+  void EmitStringLiteralDef(raw_ostream &OS, const Twine &Decl,
+                            const Twine &Indent = "  ") const {
+    OS << formatv(R"(
+{0}#ifdef __GNUC__
+{0}#pragma GCC diagnostic push
+{0}#pragma GCC diagnostic ignored "-Woverlength-strings"
+{0}#endif
+{0}{1} = {{
```

Comment:
Preprocessor directives shouldn't be indented. (And I'm not sure it is worth indenting the definition.)

Please drop the braces so that the definition looks like
`const char array[] = "initializer";`

Reply:
Right, clang-format undoes indentation for preprocess directives.

Reply:
One nit-pick here is that `char x[] = "abc"` actually creates an array of size 4, so the last `\0` is not needed I think, but I think its fine to keep the code simple.